src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -134,8 +219,241 @@ void StorageRabbitMQ::heartbeatFunc()
 
 void StorageRabbitMQ::loopingFunc()
 {
-    LOG_DEBUG(log, "Starting event looping iterations");
-    event_handler->startLoop();
+    if (event_handler->connectionRunning())
+        event_handler->startLoop();
+}
+
+
+/* Need to deactivate this way because otherwise might get a deadlock when first deactivate streaming task in shutdown and then
+ * inside streaming task try to deactivate any other task
+ */
+void StorageRabbitMQ::deactivateTask(BackgroundSchedulePool::TaskHolder & task, bool wait, bool stop_loop)
+{
+    if (stop_loop)
+        event_handler->updateLoopState(Loop::STOP);
+
+    if (task_mutex.try_lock())
+    {
+
+        task->deactivate();
```

Comment:
If an exception happened here and we will get abandoned mutex. Use unique_lock, it will unlock in the destructor.